### ISSUE TITLE
Subset of classes to port from Java to C#

### ISSUE THREAD

**Issue by smalers** [OpenCDSS/cdss-app-statemod-cs#4]
The following are a subset of StateMod Java classes to port to C# to demonstrate the prototype C# application:

* `cdss-lib-models-java`
    + `StateMod_RiverNetworkNode.java` - reads the `*.rin` file
    + `StateMod_TS` - reads and writes a StateMod time series file - focus on the read methods that are needed
    + `StateMod_StreamGage` - reads the `*.ris` file
* It will be important to port file/string utility code or find equivalent in .NET itself
    + formatting strings was done in Java with StringUtil.formatString() but there are often built-in methods now
    + StringUtil.fixedRead() may need to be ported from Java
    + padding and unpadding strings, and other string manipulations should be built in but can use old methods that just call built-in if that makes it easier
    + we need to track performance so we can compare Java and C#, see the Util.Time.StopWatch class

**Comment by jurentie** [comment]
Update:
* `.rsp` - response files can be read in and all lines are being read, effectively creating data sets with data set components for each command file. 
* `StateMod_RiverNetworkNode.java` / `*.rin` files are being read
* `StateMod_TS` files are being read.

**Comment by jurentie** [comment]
Ported many of the StateMod files over from the auto-generated code and am now reading in most StateMod file formats. I had to comment out some code that would involve porting over large amounts of files or code. For example, anything using complicated date management such as Gregorian Calendar which operates much differently in c# than in java. Therefore, in reading some file formats the date data needs to be updated. I've also ignored GeoRecord data.
